academy/4-my-own-chain/migration.md

```diff
@@ -562,131 +1057,917 @@ func getOrNewPlayerInfo(infoSoFar *map[string]*types.PlayerInfo, playerIndex str
+
+```sh
+$ checkersd query checkers list-stored-game
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Docker">
+
+```sh
+$ docker exec -it checkers checkersd query checkers list-stored-game
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+### Governance proposal
+
+For the governance proposal, you want to make sure that it stops the chain not too far in the future and after the voting period. Take 15 minutes. How many seconds does a block take?
+
+<CodeGroup>
+
+<CodeGroupItem title="Local" active>
+
+```sh
+$ jq ".app_state.mint.params.blocks_per_year" ~/.checkers/config/genesis.json
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Docker">
+
+```sh
+$ docker exec -it checkers bash -c 'jq ".app_state.mint.params.blocks_per_year" /root/.checkers/config/genesis.json'
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+This returns something like:
+
+```txt
+"6311520"
+```
+
+That many blocks per year computes down to 5 seconds per block. At this rate, 15 minutes mean 180 blocks.
+
+What is the current block height:
+
+<CodeGroup>
+
+<CodeGroupItem title="Local" active>
+
+```sh
+$ checkersd status | jq ".SyncInfo.latest_block_height"
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Docker">
+
+```sh
+$ docker exec -it checkers bash -c 'checkersd status | jq ".SyncInfo.latest_block_height"'
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+This returns something like:
+
+```json
+"1000"
+```
+
+Confirm that block issuance is indeed as you expect one every 5 seconds by running the `status` command several times. It is possible that Ignite overrides this value and issues one block per second. If that is the case, adjust the upgrade height accordingly. For example for 15 minutes:
+
+<CodeGroup>
+
+<CodeGroupItem title="At 5s block" active>
+
+```txt
+--upgrade-height 1180
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="At 1s block">
+
+```txt
+--upgrade-height 1900
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+What is the minimum deposit for a proposal:
+
+<CodeGroup>
+
+<CodeGroupItem title="Local" active>
+
+```sh
+$ jq ".app_state.gov.deposit_params.min_deposit" ~/.checkers/config/genesis.json
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Docker">
+
+```sh
+$ docker exec -it checkers bash -c 'jq ".app_state.gov.deposit_params.min_deposit" /root/.checkers/config/genesis.json' 
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+This returns something like:
+
+```json
+[
+    {
+        "amount": "10000000",
+        "denom": "stake"
+    }
+]
+```
+
+This is the amount that Alice has to deposit when submitting the proposal. For instance:
+
+```txt
+--deposit 10000000stake
+```
+
+Submit your governance proposal upgrade:
+
+<CodeGroup>
+
+<CodeGroupItem title="Local" active>
+
+```sh
+$ checkersd tx gov submit-proposal software-upgrade v1tov2 \
+    --title "Introduce Leaderboard" \
+    --description "Increase engagement via the use of a leaderboard" \
+    --from $alice \
+    --upgrade-height 1180 \
+    --deposit 10000000stake --yes
```

Comment:
@nooomski Some help perhaps because although I mention the `v1tov2` name, it does not appear after I restart (see below) with v2. This name is defined in code here https://github.com/cosmos/b9-checkers-academy-draft/blob/migration/app/app.go#L627
On this line 627, `upgradeInfo.Name` is returned as `""`. I had to panic to extract that.